pandas/io/excel.py

```diff
@@ -304,7 +304,8 @@ def read_excel(io,
                **kwds):
 
     if not isinstance(io, ExcelFile):
-        io = ExcelFile(io, engine=engine)
+        session = kwds.get('session', None)
```

Comment:
just list session as a kwarg in read_excel (and in ExcelFile), then just pass it in